drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/models/ScenarioGridModel.java

```diff
@@ -515,6 +518,58 @@ public void clearSelections() {
         selectedColumn = null;
     }
 
+    /**
+     * It forces a <code>internalRefreshWidth</code> refresh
+     */
+    public boolean forceRefreshWidth() {
+        return internalRefreshWidth(true, OptionalDouble.empty());
+    }
+
+    /**
+     * It updates a column width and its related <code>factMapping</code> columnWidth
+     * @param column
+     * @param width
+     */
+    public void updateColumnWidth(final GridColumn<?> column, final double width) {
+        column.setWidth(width);
+        synchronizeFactMappingWidth(column);
+    }
+
+    /**
+     * It synchronizes all columns related <code>factMapping</code> columnnWidths
+     */
+    public void synchronizeFactMappingsWidths() {
+        getColumns().forEach(column -> synchronizeFactMappingWidth(column));
+    }
+
+    /**
+     * It updates a column related <code>factMapping</code> columnWidth
+     * @param column
+     */
+    public void synchronizeFactMappingWidth(final GridColumn<?> column) {
+        if (!column.isVisible()) {
+            return;
+        }
+        final int columnIndex = getColumns().indexOf(column);
+        final FactMapping factMapping = simulation.getSimulationDescriptor().getFactMappingByIndex(columnIndex);
+        factMapping.setColumnWidth(column.getWidth());
+    }
+
+    /**
+     * It retrieves the stored columnWidths and assigns them to every grid column.
+     */
+    public void loadFactMappingsWidth() {
+        for (final GridColumn<?> column : getColumns()) {
+            final int columnIndex = getColumns().indexOf(column);
+            final FactMapping factMapping = simulation.getSimulationDescriptor().getFactMappingByIndex(columnIndex);
+            if (factMapping.getColumnWidth() != null) {
+                column.setWidth(factMapping.getColumnWidth());
+            } else {
```

Comment:
@yesamer 
I think this `else` branch is not needed (see above comments)

Reply:
Right, done.